### ISSUE TITLE
RVB - Realistic Damage System

### ISSUE THREAD

**Issue by Frvetz** [MathiasHun/FS22_Real_Vehicle_Breakdowns_BETA#3]
Hello MathiasHun,

I'm not sure if you know my name, but I'm pretty sure you know the RealisticDamageSystem (https://github.com/Frvetz/FS22_RealisticDamageSystem).

I am the creator of the RealisticDamageSystem and have now become aware of your script. I am so sure that you know my script because you use parts of the code without asking for my permission or even contacting me beforehand.

Even though I don't like this behaviour at all, I wanted to ask if we want to exchange contact details (Discord?) to work together and create a combination of both scripts? For example including the CVT_addon / table system for damages / DashboardLive support / tutorial / and so on

btw: still great work

Best regards 
Gerrit / frvetz

discord: frvetz

**Comment by MathiasHun** [comment]
Hi, I'm sorry. I've been writing the mod for quite some time, then maybe in January I saw you shared your mod. That's why I checked that if it's similar to what I make, I'll delete my mod. But I saw that the two mods were not similar, so I continued. I copied a time calculator, I wrote was a very long code. The motor start, motor stop sounds play similar . I wrote that part just then. After that came many tests, many errors that had to be solved. It really should have been tested on a server, so I shared it ahead of time. Unfortunately, this is how the contact failed. Of course we can work together, but I don't have Discord. I never used it. My language skills are also limited, it can only be written.
I'm currently writing from my phone

**Comment by Frvetz** [comment]
No worries, I planned something like your script for a future update but I guess now the best thing to do is to make my script and your script compatible... So a damage in my script triggers a fuction from your script. 
What do you think? I think there is a huge potential of the two mods working together!

Do you have any other form of contacting you?

Cheers

**Comment by SbSh-DNA-ModDing** [comment]
Otherwise a full fusion, addition out of both scrips?!

Our DC: https://discord.gg/uDGbaNvfgU  @MathiasHun

**Comment by haker146** [comment]
Hi :)
I'm a Farming Simulator fan and I'm really looking forward to your collaboration with @Frvetz . Is there any news yet when both mods can be combined?

**Comment by Frvetz** [comment]
Hi :) 

Haven't heard anything from him yet so I think I'll do something on my own which includes his script but can also be used without.

However I can't really tell you when that will be. I think I am going to release a version 1.0 of my script on modhub first since it runs pretty much without any bugs right now and then work on a version 2.0 with the integration :D 

Cheers
Frvetz

**Comment by Onicaro** [comment]
It's really a shame 'cause both mods have great ideas!! Have to choose between one of them sucks. Hope you gus can work something out and merge them or maybe make them compatible

**Comment by Frvetz** [comment]
From what I've found in my tests with both mods there shouldn't be any problems when playing with both of them. You can just reduce the amount of damages from the RealisticDamageSystem in the settings and your good to go :)

**Comment by JesperVVS** [comment]
Am I the only one who sees that we are missing a "jumping cables"  .. because if I now have a fellow player who thinks he just needs to be funny and turn on the light .. then the battery is dead ?  btw .. it would be great if your mods could work together  :)

**Comment by Frvetz** [comment]
;) right now you can play with both mods without compatibility issues and an integration is planned for the future. Not sure when that will be though, but it is planned

**Comment by HvyHal** [comment]
> Am I the only one who sees that we are missing a "jumping cables" .. because if I now have a fellow player who thinks he just needs to be funny and turn on the light .. then the battery is dead ? btw .. it would be great if your mods could work together :)

No you're not the only one who thinks we are missing jumper cables.  I would love to have them. Or even just pulling up another vehicle and then opening the RVB dialogue box while the good vehicle is running near in close proximity to the one with dead battery.  You don't have to have visible jumpers but you can at least simulate it in the scripting by getting the two vehicles near each other. A jumper box would work nice too you have that as a placeable that you can pick up and carry around with some kind of shop trigger on it.

**Comment by JesperVVS** [comment]
Yes or release some code .. so that it could possibly throw it into a workshop trailer ..

**Comment by HvyHal** [comment]
> ;) right now you can play with both mods without compatibility issues and an integration is planned for the future. Not sure when that will be though, but it is planned

really honestly i just want RVB by itself.  I already have other mods that im satisfied with that help slow the damages down to a more reasonable rate. I even have shelter matters which adds damage to your vehicles when they are left outside instead of stored away in a barn or shop. I just want something that makes certain components fail during old age like hundreds of hours and with high damage.

**Comment by HvyHal** [comment]
heck even thousands of hours.  Realistically most tractors don't start having major issues until they reach multiple thousands of hours.  I don't know how practical that would be in FS but you get my point I think.

**Comment by HvyHal** [comment]
> Hi :)
> 
> Haven't heard anything from him yet so I think I'll do something on my own which includes his script but can also be used without.
> 
> However I can't really tell you when that will be. I think I am going to release a version 1.0 of my script on modhub first since it runs pretty much without any bugs right now and then work on a version 2.0 with the integration :D
> 
> Cheers Frvetz

you were upset he used yours without permision but now you want to use his without permision? why is it ok for you but not the other guy?  Remind not to use your mods any more.  That's pretty dirty if you ask me.

**Comment by Frvetz** [comment]
> you were upset he used yours without permision but now you want to use his without permision? why is it ok for you but not the other guy? Remind not to use your mods any more. That's pretty dirty if you ask me.

Not at all, I think you misunderstood me. It wouldn't make any sense at all if I was mad about him using my code and then me using his in my work. That would be super dirty and would be absolutely not okay

I was talking about changing the general use case of my mod. That I'm going to go ahead with the plan I already had in mind and create an update that simulates individual damage, similar to the idea he implemented in his mod.

I wasn't mad that he released a mod that has a similar function to mine, I just didn't think it was correct that he used code and functions with complex calculations of mine, some of which took me several weeks to do, without asking. I would never use lines of code of others without obtaining the necessary permissions, which unfortunately he did. 

When I saw his mod, I thought to myself that you can combine the two mods and benefit from each other's calculations and complex tables without having to steal anything, unfortunately he didn't respond to that, whereupon I will simply continue my plan for the future on my own but without using functions/code from him. An update with individual damages (like he has) was planned from the beginning and it was also publicly known that I would do that and that's exactly what I'm going to do on my own now. I'm not mad that he published my idea before me, because he must have been working on it at least as long as me and we didn't know that someone else was working on such a mod, but if he doesn't want to work together or create two separate mods that complement each other, I don't see a problem with doing my ideas with my own code, even if the ideas of the functions are similar. My only issue was that he used complex code of mine without asking me.

I hope I could make it a bit clearer now and you understand my point of view. Whether you (continue to) use my mods or not is of course up to you

Cheers
